optimism/FunctionSpace.py

```diff
@@ -354,25 +363,43 @@ def integrate_function_on_edges(functionSpace, func, U, quadRule, edges):
     return np.sum(integrate_on_edges(functionSpace, func, U, quadRule, edges))
 
 
-class DofManager:
+class DofManager(eqx.Module):
+    # TODO get type hints below correct
+    # TODO this one could be moved to jax types if we move towards
+    # TODO jit safe preconditioners/solvers
```

Comment:
@btalamini Is this something you can help with?

Reply:
I can circle back for these type hints later on. See this for some canned solvers [lineax](https://github.com/patrick-kidger/lineax)